jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py

```diff
@@ -139,10 +139,36 @@ async def test_delete_application_no_file_uploaded(
     mocked_application_deleter.assert_called_once_with(inserted_id)
 
 
+@pytest.mark.asyncio
+async def test_delete_application__fails_if_owner_is_not_verified(
+    client, fill_application_data, inject_security_header
+):
+    """
+    Test DELETE /applications/<id> fails if the application is not owned by the requestor.
+
+    This test proves that an application is not deleted when a user that does not own it makes a DELETE
+    request to the /applications/<id> endpoint. We show this by asserting that the endpoint returns
+    a status code of 403 if the owner does not match.
+    """
+    inserted_id = await database.execute(
+        query=applications_table.insert(),
+        values=fill_application_data(application_owner_email="righ-owner@allowed.com"),
+    )
+    count = await database.fetch_all("SELECT COUNT(*) FROM applications")
+    assert count[0][0] == 1
```

Comment:
so, as I said somewhere in other PR: this test doesn't guarantee that the application in the database is the one you want to. [Here](https://github.com/omnivector-solutions/notifications/blob/main/notifications-api/tests/routers/test_graphql_router.py#L317-L339) I paste an example of a test that guarantees such a thing.

Reply:
Here I am following the pattern used throughout the Jobbergate API tests. It is verifying that a row was inserted to prepare the test. I find these assertions before the test logic pretty useless, but I didn't want to change the entire test code for this.

Doing this sort of assertion _before the code under test_ is actually very inefficient and a waste of queries. If the `execute` command that you are running to prepare the testbed completes successfully, you should assume the rows were inserted. This is especially true since execute is returning the new row id from the insertion.

I will create an issue to clean up the tests in Jobbergate API to remove the redundant assertions and make sure that the assertions following the code under test are in fact checking that the correct values are in the database.